xdsl/dialects/riscv.py

```diff
@@ -2492,9 +2529,593 @@ class RemuOp(RdRsRsIntegerOperation[IntRegisterType, IntRegisterType]):
+    X(rd) = result;
+    ```
+    See external https://five-embeddev.com/riscv-bitmanip/1.0.0/bitmanip.html#insns-rori
+    """
+    name = "riscv.rori"
+
+    traits = traits_def(Pure())
+
+@irdl_op_definition
+class RoriwOp(RdRsImmShiftOperation):
+    """
+    This instruction performs a rotate right on the least-significant word of rs1 by the amount in 
+    the least-significant log2(XLEN) bits of shamt. The resulting word value is sign-extended by 
+    copying bit 31 to all of the more-significant bits.
+    
+    ```
+    let rs1 = EXTZ(X(rs1)[31..0];
+    let result = (rs1 >> shamt[4..0]) | (X(rs1) << (32 - shamt[4..0]));
+    X(rd) = EXTS(result[31..0]);
+    ```
+    See external https://five-embeddev.com/riscv-bitmanip/1.0.0/bitmanip.html#insns-roriw
+    """
+    name = "riscv.roriw"
+
+    traits = traits_def(Pure())
+
+@irdl_op_definition
+class AddUwOp(RdRsRsIntegerOperation[IntRegisterType, IntRegisterType]):
+    """
+    This instruction performs an XLEN-wide addition between rs2 and the zero-extended least-significant 
+    word of rs1.
+    
+    ```
+   let base = X(rs2);
+   let index = EXTZ(X(rs1)[31..0]);
+   X(rd) = base + index;
+    ```
+    See external https://five-embeddev.com/riscv-bitmanip/1.0.0/bitmanip.html#insns-add_uw
+    """
+    name = "riscv.add.uw"
+
+    traits = traits_def(Pure())
+
+@irdl_op_definition
+class Sh1addOp(RdRsRsIntegerOperation[IntRegisterType, IntRegisterType]):
+    """
+    This instruction shifts rs1 to the left by 1 bit and adds it to rs2.
+    
+    ```
+   X(rd) = X(rs2) + (X(rs1) << 1);
+    ```
+    See external https://five-embeddev.com/riscv-bitmanip/1.0.0/bitmanip.html#insns-sh1add
+    """
+    name = "riscv.sh1add"
+
+    traits = traits_def(Pure())
+
+@irdl_op_definition
+class Sh2addOp(RdRsRsIntegerOperation[IntRegisterType, IntRegisterType]):
+    """
+    This instruction shifts rs1 to the left by 2 places and adds it to rs2.
+    
+    ```
+   X(rd) = X(rs2) + (X(rs1) << 2);
+    ```
+    See external https://five-embeddev.com/riscv-bitmanip/1.0.0/bitmanip.html#insns-sh2add
+    """
+    name = "riscv.sh2add"
+
+    traits = traits_def(Pure())
+
+@irdl_op_definition
+class Sh3addOp(RdRsRsIntegerOperation[IntRegisterType, IntRegisterType]):
+    """
+    This instruction shifts rs1 to the left by 2 places and adds it to rs2.
+    
+    ```
+   X(rd) = X(rs2) + (X(rs1) << 3);
+    ```
+    See external https://five-embeddev.com/riscv-bitmanip/1.0.0/bitmanip.html#insns-sh3add
+    """
+    name = "riscv.sh3add"
+
+    traits = traits_def(Pure())
+
+@irdl_op_definition
+class Sh1addUwOp(RdRsRsIntegerOperation[IntRegisterType, IntRegisterType]):
+    """
+    This instruction performs an XLEN-wide addition of two addends. The first addend is rs2. 
+    The second addend is the unsigned value formed by extracting the least-significant word of 
+    rs1 and shifting it left by 1 place.
+    
+    ```
+   let base = x[rs2];
```

Comment:
This is more of a nit, but I know this will frustrate me until it's fixed, could you please indent all blocks to four spaces?